rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -76,13 +140,34 @@ error_code cellSearchInitialize(CellSearchMode mode, u32 container, vm::ptr<Cell
 
 error_code cellSearchFinalize()
 {
-	cellSearch.warning("cellSearchFinalize()");
+	cellSearch.todo("cellSearchFinalize()");
 
 	const auto search = g_fxo->get<search_info>();
 
+	switch (search->state.compare_and_swap(search_state::idle, search_state::finalizing))
+	{
+	case search_state::idle:
+		break;
+	case search_state::not_initialized:
+		return CELL_SEARCH_ERROR_NOT_INITIALIZED;
+	case search_state::finalizing:
+		return CELL_SEARCH_ERROR_FINALIZING;
+	case search_state::in_progress:
+	case search_state::initializing:
+	case search_state::canceling:
+		return CELL_SEARCH_ERROR_BUSY;
+	default:
+		return CELL_SEARCH_ERROR_GENERIC;
+	}
+
 	sysutil_register_cb([=](ppu_thread& ppu) -> s32
 	{
+		{
+			std::lock_guard lock(search->links_mutex);
+			search->content_links.clear();
```

Comment:
```suggestion
		std::lock_guard{search->links_mutex}, search->content_links.clear();
```
Only a suggestion, whatever style you prefer.

Reply:
Nah, not a fan of it